src/main/java/org/broadinstitute/consent/http/resources/DatasetResource.java

```diff
@@ -353,6 +354,20 @@ public Response validateDatasetName(@QueryParam("name") String name) {
         }
     }
 
+    @GET
+    @Consumes("application/json")
+    @Produces("application/json")
+    @Path("/studyNames")
+    @PermitAll
+    public Response findAllActiveStudyNames() {
+        try {
+            Set<String> studyNames = datasetService.findAllActiveStudyNames();
+            return Response.ok(studyNames).build();
+        } catch (Exception e) {
+            throw new NotFoundException("Could not get all study names");
```

Comment:
If you hit an exception here, there's something else wrong, this doesn't seem like a 404 to me, perhaps a 500 error since we should never have an error case here. An empty list might still be a valid list of study names in use. Also, see how we use the super method of `createExceptionResponse(e)` which will try to examine the exception type and return an appropriate error code.

Reply:
Oops yeah, this was all copied over code and schema. My bad, I should have looked more carefully at it.